### DISCUSSION TITLE
Best way to show input validation errors?

### DISCUSSION THREAD

**Discussion by agoldis** [zeit-ui/react#490] [Q&A] [answered]
Looking for recommendations on showing input validation errors. Awesome project! Thank you!

Is it this one?
<img width="292" alt="Screen Shot 2021-03-24 at 11 44 38 PM" src="https://user-images.githubusercontent.com/1637928/112430132-f11f3900-8cfa-11eb-8483-9ce0c0e1723e.png">

**Comment by unix** [comment, marked as answer]
I always use the Text component to manually display the error myself, a little annoying indeed. I think the `Input` component needs a `Label` or something to display some text information, similar to:
```
<Input>
  <Input.Label type="error">Something doesn't seem right.</Input.Label>
</Input>

<Input>
  <Input.Label type="error"><Image ... /></Input.Label>
</Input>

// custom type or not set
<Input>
  <Input.Label type="myLabel">...</Input.Label>
</Input>


// more complex combinations
<Input>
  <Input.Label type="myLabel">
    <Text h1>....</Text>
    <Text h1>....</Text>
    <Button>Try again</Button>
  </Input.Label>
</Input>
 ```


This is just a preliminary idea, if anyone would like to implement it, please create an issue first to discuss the details of the implementation. (If no one wants to contribute this, I will work on it when I have time in the next few weeks)

**Reply by unix** [reply]
This feature is very useful, I often suffer from this, too. But it's not easy to make it as good as it can be, and anyone with ideas can feel free to comment.
